CIP-XXXX/README.md

```diff
@@ -0,0 +1,190 @@
+
+### Extension CIPs
+
+Extension authors should strive to construct self-contained specifications that do not overlap with other extensions, in scope, data types, error codes, or endpoints. If there is overlap, then it is at the implementors discretion to merge as they see fit. To attempt to avoid this, elements of extensions should be scope specific.
+
+The creation of competing extensions scopes should be avoided; rather, extension scopes should be clear, non-ambiguous, and siloed. Ideally, the scope should be an attempt to solve a problem or address unrealized functionality in the set of dApp connector extensions.
+
+Versioning of extensions should be avoided, once accepted through the CIP process extension's functionality should be static. If an extension needs substantial updates replacing it with a successor is preferred. Extensions replacing older extensions should clearly state their intent to do so.
+
+##### Can extensions add their own Data Types, Error Types and Error Codes?
+
+Yes.
+
+##### Can extensions add their own endpoints?
+
+Yes. Endpoints can be added to this proposal's [Extendable API](#extendable-api), but not the [Initial API](#initial-api).
+
+##### Can extensions overwrite/replace any part of this specification?
+
+No.
+
+##### Can extensions depend on other extensions?
+
+No. Currently we do not allow this.
+
+##### Should extensions follow a specific format?
+
+Yes. They all are CIPs.
+
+##### Are wallet expected to implement all extensions?
+
+No. It's up to wallet providers to decide which extensions they ought to support.
+
+### Error Types
+
+#### ConnectorError
+
+```js
+ConnectorErrorCode {
+	InvalidRequest: -1,
+	InternalError: -2,
+	Refused: -3,
+}
+ConnectorError {
+	code: ConnectorErrorCode,
+	info: string
+}
+```
+
+* InvalidRequest - Arguments do not conform to this specification or are otherwise invalid - e.g. requested extension is not supported by the wallet.
+* InternalError - An error occurred during execution of this API call.
+* Refused - The request was refused due to lack of access - e.g. user rejected connection to this dApp.
+
+### Initial API
+
+In order to initiate communication from webpages to a user's Cardano wallet, the wallet must provide the following javascript API to the webpage. A shared, namespaced `cardano` object must be injected into the page if it did not exist already. Each wallet implementing this standard must then create a field in this object with a name unique to each wallet containing a `wallet` object with the following methods. The API is split into two stages to maintain the user's privacy, as the user will have to consent to `cardano.{walletName}.enable()` in order for the dApp to read any information pertaining to the user's wallet with `{walletName}` corresponding to the wallet's namespaced name of its choice.
+
+#### cardano.{walletName}.enable(extensions: Extension[] = [{ cip: 30 }]): Promise\<API>

```

Comment:
Same comment about structural vs positionsl as I left on the other CIP30 PR

Reply:
Given this is defined in CIP-30 and is already implemented by wallets, this is asking for trouble on a backwards compatibility front.

It would be simpler if the wallets just `enable` ALL extensions they are aware of.

Alternatively, there could be:
```
cardano.{wallet}.reqExtensions(extensions: Extension[])
```
This could pre-informs the wallet what extensions the dapp wants to use.  The `enable` then just `enables` those.  The default if its not called should be enable all extensions the wallet supports.  However, in this case, I think its informative only (maybe the wallet can use it to display something meaningful to the user?).  The wallet should enable all the extensions that it supports, which were requested.  If an extension was requested but the wallet does not support it, that should not be an error.